src/rgw/rgw_reshard.cc

```diff
@@ -451,22 +505,32 @@ int RGWBucketReshard::do_reshard(
 
 	int shard_index = (target_shard_id > 0 ? target_shard_id : 0);
 
-	ret = target_shards_mgr.add_entry(shard_index, entry, account, category, stats);
+	ret = target_shards_mgr.add_entry(shard_index, entry, account,
+					  category, stats);
 	if (ret < 0) {
 	  return ret;
 	}
+
+	if (Clock::now() >= lock_renew_thresh) {
+	  ret = renew_lock_bucket();
```

Comment:
consider passing the value of `now` into `renew_lock_bucket()` so it doesn't have to immediately re-sample the Clock. i like how the renew_locks_callback() does this

Reply:
yes, I didn't notice this, but if renew_lock_bucket() is going to do it again...